shell/browser/electron_browser_client.cc

```diff
@@ -1515,14 +1515,23 @@ void ElectronBrowserClient::
             render_frame_host,  // NOLINT(runtime/references)
         blink::AssociatedInterfaceRegistry&
             associated_registry) {  // NOLINT(runtime/references)
-  associated_registry.AddInterface(base::BindRepeating(
-      [](content::RenderFrameHost* render_frame_host,
-         mojo::PendingAssociatedReceiver<electron::mojom::ElectronBrowser>
-             receiver) {
-        ElectronBrowserHandlerImpl::Create(render_frame_host,
-                                           std::move(receiver));
-      },
-      &render_frame_host));
+  auto* contents =
+      content::WebContents::FromRenderFrameHost(&render_frame_host);
+  if (contents) {
+    auto* prefs = WebContentsPreferences::From(contents);
+    if (render_frame_host.IsInPrimaryMainFrame() ||
+        (prefs && prefs->AllowsNodeIntegrationInSubFrames())) {
+      associated_registry.AddInterface(base::BindRepeating(
```

Comment:
Had a thought since our last conversation about this. There are some non-JS-accessible IPCs that the `ElectronBrowser` interface provides:

```
  OnFirstNonEmptyLayout();

  UpdateDraggableRegions(
    array<DraggableRegion> regions);

  SetTemporaryZoomLevel(double zoom_level);

  [Sync]
  DoGetZoomLevel() => (double result);
```

Do we need any of those on non-main frames?

Reply:
@nornagon Maybe we should split those out into a separate mojo interface?

I think they are all legitimately needed in some cases 🤔

Reply:
yeah i wouldn't be averse to splitting these out. Maybe these ones can stay as `ElectronBrowser` and the rest can go in `ElectronBrowserIpc` or something?